packages/itwinui-react/src/core/utils/components/Flex.tsx

```diff
@@ -0,0 +1,271 @@
+
+const getValueForToken = (token?: string) => {
+  if (sizeTokens.includes(token as SizeToken)) {
+    return `var(--iui-size-${token})`;
+  }
+  return token;
+};
+
+// ----------------------------------------------------------------------------
+// Main Flex component
+
+const FlexComponent = React.forwardRef((props, ref) => {
+  const {
+    as: Element = 'div',
+    display,
+    flexDirection,
+    justifyContent,
+    alignItems,
+    gap,
+    flexWrap,
+    className,
+    style,
+    ...rest
+  } = props;
+
+  return (
+    <Element
+      className={cx('iui-flex', className)}
+      style={
+        {
+          '--iui-flex-display': display,
+          '--iui-flex-direction': flexDirection,
+          '--iui-flex-justify': justifyContent,
+          '--iui-flex-align': alignItems,
+          '--iui-flex-gap': getValueForToken(gap),
+          '--iui-flex-wrap': flexWrap,
+          ...style,
+        } as React.CSSProperties
+      }
+      ref={ref}
+      {...rest}
+    />
+  );
+}) as PolymorphicForwardRefComponent<'div', FlexOwnProps>;
+type FlexOwnProps = {
+  /**
+   * Value of the `display` property.
+   * @default 'flex'
+   */
+  display?: 'flex' | 'inline-flex';
+  /**
+   * Value of the `justify-content` property.
+   * @default 'flex-start'
+   */
+  justifyContent?: React.CSSProperties['justifyContent'];
+  /**
+   * Value of the `align-items` property. Defaults to 'center' but you
+   * might want to use 'flex-start' in some cases.
+   * @default 'center'
+   */
+  alignItems?: React.CSSProperties['alignItems'];
+  /**
+   * Value of the `gap` property.
+   *
+   * Supports aliases for iTwinUI size tokens:
+   * 3xs, 2xs, xs, s, m, l, xl, 2xl, 3xl
+   *
+   * Also accepts any valid css value (e.g. "1rem" or "2px")
+   *
+   * @default '2xs'
+   */
+  gap?: SizeToken | AnyString;
+  /**
+   * Value of the `direction` property.
+   * @default 'row'
+   */
+  flexDirection?: React.CSSProperties['flexDirection'];
+  /**
+   * Value of the `flex-wrap` property.
+   * @default 'nowrap'
+   */
+  flexWrap?: React.CSSProperties['flexWrap'];
+};
+
+// ----------------------------------------------------------------------------
+// Flex.Spacer component
+
+const FlexSpacer = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', flex, className, style, ...rest } = props;
+  return (
+    <Element
+      className={cx('iui-flex-spacer', className)}
+      style={
+        { '--iui-flex-spacer-flex': flex, ...style } as React.CSSProperties
+      }
+      ref={ref}
+      {...rest}
+    />
+  );
+}) as PolymorphicForwardRefComponent<'div', FlexSpacerOwnProps>;
+type FlexSpacerOwnProps = {
+  /**
+   * Value of the `flex` css property.
+   *
+   * This may need to be adjusted if you're also setting `flex` on
+   * other `Flex.Item`s.
+   *
+   * @default 999
+   */
+  flex?: React.CSSProperties['flex'];
+};
+
+// ----------------------------------------------------------------------------
+// Flex.Item subcomponent
```

Comment:
I kept everything in the same file and added these dividers.

For me this felt much easier to navigate and read, vs having every subcomponent in a separate file.

I can split into separate files if others feel differently.